### ISSUE TITLE
OSRM routed thread

### ISSUE THREAD

**Issue by helseb** [Project-OSRM/osrm-backend#1963]
Hello,

When launching a request on table service api, even if my orsm-routed is launched with multi threads option, it seems that only one core is working when processing.
Is it normal?

Command used : `osrm-routed --shared-memory=yes --threads=8`

**Comment by MoKob** [comment]
Using multiple threads only handles multiple requests in parallel. Every single request is served by a single thread.

You could try and split it into subtask, but the speed-up of such a separation would suffer from the way the algorithm works.
Computing two tables of size (rows/2 * cols) would require O(rows/2 + cols) work, so you could at most a reduction to 75% of running time using two threads (50% + 50/threads) in an ideal world.
This way of computing the tables in parallel requires multiple requests and an aggregation into a big table on your end.
